****Ensemble skills work the same way as normal skills. 
****Any two ensembles stack
****Allow all skills while singing if longing
****Throw Arrow/Musical Strike? (okay as is, class's only non-reborn attack skill, really.)

****Longing for Freedom -> Allow skills and attacks during song

****Frost Joke/Scream delay->Cooldown

****Bug : Song got interrupted after Longing!





diff --git a/db/re/skill_cast_db.txt b/db/re/skill_cast_db.txt
index b872350dae..1f5d1fbce2 100644
--- a/db/re/skill_cast_db.txt
+++ b/db/re/skill_cast_db.txt

diff --git a/db/re/skill_db.txt b/db/re/skill_db.txt
index 72742754ec..808b9d5fcd 100644
--- a/db/re/skill_db.txt
+++ b/db/re/skill_db.txt


diff --git a/db/re/skill_nocast_db.txt b/db/re/skill_nocast_db.txt
index 6d4213ef4e..43a1de57e1 100644
--- a/db/re/skill_nocast_db.txt
+++ b/db/re/skill_nocast_db.txt
@@ -26,6 +26,11 @@
 //----------------------------------------------------------------------------
 // Normal
 //----------------------------------------------------------------------------
+308,1	//BD_ETERNALCHAOS
+311,1	//BD_ROKISWEIL
+317,1	//BA_DISSONANCE
+325,1	//DC_UGLYDANCE
+328,1	//DC_DONTFORGETME
 2422,1	//WM_LULLABY_DEEPSLEEP
 2423,1	//WM_SIRCLEOFNATURE
 2430,1	//WM_SATURDAY_NIGHT_FEVER
diff --git a/db/re/skill_require_db.txt b/db/re/skill_require_db.txt
index 15c7a09627..9a8247a222 100644
--- a/db/re/skill_require_db.txt
+++ b/db/re/skill_require_db.txt

@@ -541,7 +541,7 @@
 1008,0,0,30,0,0,0,99,0,0,none,0,0,12115,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0							//SA_ELEMENTWATER
 1009,0,0,10,0,0,0,11,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0								//HT_PHANTASMIC
 1010,0,0,20,0,0,0,99,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0								//BA_PANGVOICE
-1011,0,0,40,0,0,0,99,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0								//DC_WINKCHARM
+1011,0,0,20,0,0,0,99,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0								//DC_WINKCHARM
 1013,0,0,10,0,0,0,99,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0								//BS_GREED
 1014,0,0,400,0,0,0,99,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0							//PR_REDEMPTIO
 1015,0,0,40,0,0,0,99,0,0,none,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0								//MO_KITRANSLATION
diff --git a/src/map/battle.cpp b/src/map/battle.cpp
index de8167f4a0..58c201f9a2 100644
--- a/src/map/battle.cpp
+++ b/src/map/battle.cpp
@@ -5950,22 +6065,38 @@ struct Damage battle_calc_magic_attack(struct block_list *src,struct block_list
...
 							skillratio += 20 * skill_lv - 20; //Monsters use old formula
-						}
+						break;
+					case BA_DISSONANCE:
+						skillratio += skill_lv * 10;
+						if (sd)
+							skillratio += 3 * pc_checkskill(sd, BA_MUSICALLESSON);
+						break;
 					case WZ_VERMILION:
 						skillratio += 20 * skill_lv - 20;
 						break;
@@ -6339,12 +6467,17 @@ struct Damage battle_calc_magic_attack(struct block_list *src,struct block_list
 					struct Damage wd = battle_calc_weapon_attack(src,target,skill_id,skill_lv,mflag);
 
 					ad.damage = battle_attr_fix(src, target, wd.damage + ad.damage, s_ele, tstatus->def_ele, tstatus->ele_lv) * (100 + 40 * skill_lv) / 100;
+#ifdef RENEWAL
+					if (src == target)
+						ad.damage = 0;
+#else
 					if(src == target) {
 						if(src->type == BL_PC)
 							ad.damage = ad.damage / 2;
 						else
 							ad.damage = 0;
 					}
+#endif
 				}
 				break;
 			case SO_VARETYR_SPEAR: {
@@ -6487,11 +6625,13 @@ struct Damage battle_calc_misc_attack(struct block_list *src,struct block_list *
 				}
 			}
 			break;
+#ifndef RENEWAL
 		case BA_DISSONANCE:
 			md.damage = 30 + skill_lv * 10;
 			if (sd)
 				md.damage += 3 * pc_checkskill(sd,BA_MUSICALLESSON);
 			break;
+#endif
 		case NPC_SELFDESTRUCTION:
 			md.damage = sstatus->hp;
 			break;
diff --git a/src/map/clif.cpp b/src/map/clif.cpp
index c3cb4a2b0e..f7907a3897 100644
--- a/src/map/clif.cpp
+++ b/src/map/clif.cpp
diff --git a/src/map/map-server.vcxproj b/src/map/map-server.vcxproj
index dd1c315170..f448f9df44 100644
--- a/src/map/map-server.vcxproj
+++ b/src/map/map-server.vcxproj
@@ -355,7 +355,7 @@
     <Copy SourceFiles="$(SolutionDir)db\import-tmpl\produce_db.txt" DestinationFolder="$(SolutionDir)db\import\" ContinueOnError="true" Condition="!Exists('$(SolutionDir)db\import\produce_db.txt')" />
     <Copy SourceFiles="$(SolutionDir)db\import-tmpl\quest_db.txt" DestinationFolder="$(SolutionDir)db\import\" ContinueOnError="true" Condition="!Exists('$(SolutionDir)db\import\quest_db.txt')" />
     <Copy SourceFiles="$(SolutionDir)db\import-tmpl\refine_db.yml" DestinationFolder="$(SolutionDir)db\import\" ContinueOnError="true" Condition="!Exists('$(SolutionDir)db\import\refine_db.yml')" />
-    <Copy SourceFiles="$(SolutionDir)db\import-tmpl\size_fix.txt" DestinationFolder="$(SolutionDir)db\import\" ContinueOnError="true" Condition="!Exists('$(SolutionDir)db\import\size_fix.txt')" />
+    <Copy SourceFiles="$(SolutionDir)db\import-tmpl\size_fix.yml" DestinationFolder="$(SolutionDir)db\import\" ContinueOnError="true" Condition="!Exists('$(SolutionDir)db\import\size_fix.yml')" />
     <Copy SourceFiles="$(SolutionDir)db\import-tmpl\skill_cast_db.txt" DestinationFolder="$(SolutionDir)db\import\" ContinueOnError="true" Condition="!Exists('$(SolutionDir)db\import\skill_cast_db.txt')" />
     <Copy SourceFiles="$(SolutionDir)db\import-tmpl\skill_castnodex_db.txt" DestinationFolder="$(SolutionDir)db\import\" ContinueOnError="true" Condition="!Exists('$(SolutionDir)db\import\skill_castnodex_db.txt')" />
     <Copy SourceFiles="$(SolutionDir)db\import-tmpl\skill_changematerial_db.txt" DestinationFolder="$(SolutionDir)db\import\" ContinueOnError="true" Condition="!Exists('$(SolutionDir)db\import\skill_changematerial_db.txt')" />
diff --git a/src/map/pc.cpp b/src/map/pc.cpp
index 9df9658861..f9c601960a 100755
--- a/src/map/pc.cpp
+++ b/src/map/pc.cpp
@@ -8844,6 +8844,11 @@ int pc_itemheal(struct map_session_data *sd, int itemid, int hp, int sp)
 			sp += sp / 10;
 		}
 
+#ifdef RENEWAL
+		if (sd->sc.data[SC_APPLEIDUN])
+			hp += sd->sc.data[SC_APPLEIDUN]->val3 / 100;
+#endif
+
 		if (penalty > 0) {
 			hp -= hp * penalty / 100;
 			sp -= sp * penalty / 100;

diff --git a/src/map/script_constants.hpp b/src/map/script_constants.hpp
index 94644b2e1a..1bae33dcd8 100644
--- a/src/map/script_constants.hpp
+++ b/src/map/script_constants.hpp
@@ -1524,6 +1524,8 @@
 	export_constant(SC_LHZ_DUN_N3);
 	export_constant(SC_LHZ_DUN_N4);
 	export_constant(SC_ANCILLA);
+	export_constant(SC_ADAPTATION);
 #ifdef RENEWAL
 	export_constant(SC_EXTREMITYFIST2);
 #endif
